### ISSUE TITLE
System.IndexOutOfRangeException on MimeMessage.Load

### ISSUE THREAD

**Issue by ekalchev** [jstedfast/MailKit#780]
I am getting the following exception when trying to parse a mime message stream. Unfortunately we don't have the email that is causing this issue. 

System.IndexOutOfRangeException: Index was outside the bounds of the array.
at MimeKit.InternetAddress.TryParseMailbox(ParserOptions options, Byte[] text, Int32 startIndex, Int32& index, Int32 endIndex, String name, Int32 codepage, Boolean throwOnError, InternetAddress& address)
at MimeKit.InternetAddress.TryParse(ParserOptions options, Byte[] text, Int32& index, Int32 endIndex, Int32 groupDepth, AddressParserFlags flags, InternetAddress& address)
at MimeKit.MimeEntity.OnHeadersChanged(HeaderListChangedAction action, Header header)
at MimeKit.MimePart.OnHeadersChanged(HeaderListChangedAction action, Header header)
at MimeKit.MimeEntity.HeadersChanged(Object sender, HeaderListChangedEventArgs e)
at MimeKit.HeaderList.OnChanged(Header header, HeaderListChangedAction action)
at MimeKit.HeaderList.Add(Header header)
at MimeKit.MimeEntity..ctor(MimeEntityConstructorArgs args)
at MimeKit.ParserOptions.CreateEntity(ContentType contentType, IList`1 headers, Boolean toplevel)
at MimeKit.MimeParser.MultipartScanSubparts(Multipart multipart, Byte* inbuf, CancellationToken cancellationToken)
at MimeKit.MimeParser.ConstructMultipart(Multipart multipart, Byte* inbuf, CancellationToken cancellationToken)
at MimeKit.MimeParser.ParseMessage(Byte* inbuf, CancellationToken cancellationToken)
at MimeKit.MimeParser.ParseMessage(CancellationToken cancellationToken)
at MimeKit.MimeMessage.Load(ParserOptions options, Stream stream, Boolean persistent, CancellationToken cancellationToken)

**Comment by jstedfast** [comment]
What version of MimeKit are you using? I don't see any obvious ways of getting this exception in the latest code.

Without an example message, I'm not sure if I'm going to be able to fix this.

**Comment by jstedfast** [comment]
A few commits after that MimeKit commit:

```
commit 4044269ce078c7721eb0c2b1c790c70e0f40af70
Author: Jeffrey Stedfast <jestedfa@microsoft.com>
Date:   Thu Jul 26 15:42:27 2018 -0400

    Add more bounds checking for TryParseMailbox()
    
    Fixes issue #421

```

:-)
